Generals/Code/GameEngine/Source/GameLogic/Object/Behavior/PropagandaTowerBehavior.cpp

```diff
@@ -205,7 +205,15 @@ UpdateSleepTime PropagandaTowerBehavior::update( void )
 		}
 	}
 
-	if( self->getContainedBy()  &&  self->getContainedBy()->getContainedBy() )
+#if RETAIL_COMPATIBLE_CRC
+	Bool contained = self->getContainedBy() && self->getContainedBy()->getContainedBy();
+#else
+	Bool contained = (self->isKindOf(KINDOF_PORTABLE_STRUCTURE)) ?
```

Comment:
Can we implement this more generic, without kindof test? I expect in its current form there could be edge cases where this check fails.

Reply:
This change does cover all cases in its current form, given that objects of type `PORTABLE_STRUCTURE` are always considered contained.

There may be an overloaded function somewhere that portable structures have to distinguish them from other passengers that could be used here. Will need further investigation.